subversion/src/org/netbeans/modules/subversion/client/cli/commands/VersionCommand.java

```diff
@@ -105,13 +104,144 @@ public boolean isUnsupportedVersion() {
     public boolean isMetadataFormatSupported() {
         return supportedMetadataFormat;
     }
-    
+
     public String getOutput() {
         StringBuffer sb = new StringBuffer();
         for (String string : output) {
             sb.append(string);
             sb.append('\n');
         }
         return sb.toString();
-    }    
+    }
+
+    /**
+     * Version holder with semantic verioning support. Keep svn --version output
+     * is in mind. The container is compatible with MAJOR.MINOR.PATCH formatp
+     * lus some remainder as it happens in svn --version. I.e. "svn, version
+     * 1.10.0 (r1827917)" is supported Regular 1.10.1 is supported too.
+     *
+     * Note on exceptions: Exceptions are unchecked and not expected to occur,
+     * if that happens its a bug.
+     */
+    public static class Version implements Comparable<Version> {
+
+        private int[] parts;
+        public String remainder;
+
+        static final Version UNKNOWN = new Version();
+
+        private Version() {
+        }
+
+        /**
+         * Parse version string into container. String must be in
+         * "MAJOR.MINOR.PATCH reminder" format. Reminder part is optional.
+         *
+         * @param version non null string
+         * @return parsed version or UKNOWN if parsing error happens
+         */
+        public static Version parse(String version) {
+            return parse(version, " ");
+        }
+
+        /**
+         * Parse version string into container. String must be in
+         * "MAJOR.MINOR.PATCH-reminder" format. Reminder part is optional.
+         * @param version string with version, say 1.1.1-SNAPSHOT
+         * @param remainderDelimiter a symbol to separate semantic version from reminder, say "-" for "1.1.1-SNAPSHOT"
+         * @return
+         */
+        public static Version parse(String version, String remainderDelimiter) {
+            String symanticVersion;
```

Comment:
Typo? I think you meant `semanticVersion`. The whole `parse` function can be simplified to:

```java
        public static Version parse(String version, String remainderDelimiter) {
            String[] versionMajorParts = version.split(Pattern.quote(remainderDelimiter), 2);

            String[] stringParts = versionMajorParts[0].split("\\.");
            int[] parts = new int[stringParts.length];
            for(int i = 0; i < stringParts.length; i++) {
                parts[i] = Integer.parseInt(stringParts[i]);
            }

            return new Version(parts, versionMajorParts.length > 1 ? versionMajorParts[1] : "");
        }
```

The `Version` attributes can be final, if they are initialized in the contructor:

```java
        private Version(int[] parts, String remainder) {
            if(parts == null) {
                throw new NullPointerException("parts must not be null");
            }
            if(remainder == null) {
                throw new NullPointerException("remainder must not be null");
            }
            this.parts = parts;
            this.remainder = remainder;
        }
```

And `trim` becomes:

```java
      private Version trim(int level) {
            if (isUnknown()) {
                return this;
            } else {
                int[] newParts = new int[level];
                System.arraycopy(parts, 0, newParts, 0, Math.min(level, parts.length));
                return new Version(newParts, this.remainder);
            }
        }
```